src/solvers/dgsem_tree/subcell_limiters.jl

```diff
@@ -41,62 +43,90 @@ The bounds are calculated using the low-order FV solution. The positivity limite
     This is an experimental feature and may change in future releases.
 """
 struct SubcellLimiterIDP{RealT <: Real, Cache} <: AbstractSubcellLimiter
+    local_minmax::Bool
+    local_minmax_variables_cons::Vector{Int}                   # Local mininum/maximum principles for conservative variables
     positivity::Bool
     positivity_variables_cons::Vector{Int}                     # Positivity for conservative variables
     positivity_correction_factor::RealT
     cache::Cache
 end
 
-# this method is used when the indicator is constructed as for shock-capturing volume integrals
+# this method is used when the limiter is constructed as for shock-capturing volume integrals
 function SubcellLimiterIDP(equations::AbstractEquations, basis;
+                           local_minmax_variables_cons = [],
                            positivity_variables_cons = [],
                            positivity_correction_factor = 0.1)
+    local_minmax = (length(local_minmax_variables_cons) > 0)
     positivity = (length(positivity_variables_cons) > 0)
 
-    bound_keys = Tuple(Symbol(string(v), "_min") for v in positivity_variables_cons)
+    bound_keys = ()
+    if local_minmax
+        for v in local_minmax_variables_cons
+            v_string = string(v)
+            bound_keys = (bound_keys..., Symbol(v_string, "_min"),
+                          Symbol(v_string, "_max"))
+        end
+    end
+    for v in positivity_variables_cons
+        if !(v in local_minmax_variables_cons)
+            bound_keys = (bound_keys..., Symbol(string(v), "_min"))
+        end
+    end
 
     cache = create_cache(SubcellLimiterIDP, equations, basis, bound_keys)
 
-    SubcellLimiterIDP{typeof(positivity_correction_factor), typeof(cache)}(positivity,
-                                                                           positivity_variables_cons,
-                                                                           positivity_correction_factor,
-                                                                           cache)
+    SubcellLimiterIDP{typeof(positivity_correction_factor),
+                      typeof(cache)}(local_minmax, local_minmax_variables_cons,
+                                     positivity, positivity_variables_cons,
+                                     positivity_correction_factor, cache)
 end
 
 function Base.show(io::IO, limiter::SubcellLimiterIDP)
     @nospecialize limiter # reduce precompilation time
-    @unpack positivity = limiter
+    (; local_minmax, positivity) = limiter
 
     print(io, "SubcellLimiterIDP(")
-    if !(positivity)
+    if !(local_minmax || positivity)
         print(io, "No limiter selected => pure DG method")
     else
         print(io, "limiter=(")
+        local_minmax && print(io, "min/max limiting, ")
         positivity && print(io, "positivity")
         print(io, "), ")
     end
+    print(io, "Local bounds with FV solution")
```

Comment:
Is this statement here intentional?

Reply:
Same as below. Since the bar states are added within the next weeks, I will keep it as it is.